lib/avalanche.ex

```diff
@@ -194,14 +194,27 @@ defmodule Avalanche do
   """
   @spec status(String.t(), keyword(), keyword()) :: any() | {:error, Avalanche.Error.t()}
   def status(statement_handle, status_options \\ [], request_options \\ []) do
-    with request_opts <- Keyword.merge(default_options(), request_options),
-         {:ok, valid_request_opts} <- validate_options(request_opts, @request_options_schema),
-         {:ok, valid_status_opts} <- validate_options(status_options, @status_options_schema),
-         async <- Keyword.fetch!(valid_status_opts, :async),
-         partition <- Keyword.fetch!(valid_status_opts, :partition) do
-      statement_handle
-      |> Avalanche.StatusRequest.build(valid_request_opts)
-      |> Avalanche.StatusRequest.run(async, partition)
+    start_time = System.monotonic_time()
+
+    metadata =
+      %{statement_handle: statement_handle}
+      |> Map.put(:async, Keyword.get(status_options, :async))
+      |> Map.put(:partition, Keyword.get(status_options, :partition))
```

Comment:
Sorry to be a pain, but maybe put this `metadata` stuff in the catch since it isn't needed anywhere else.

Reply:
haha, never a pain! Done!